src/libcore/ptr.rs

```diff
@@ -532,6 +532,15 @@ pub struct Unique<T: ?Sized> {
     _marker: PhantomData<T>,
 }
 
+#[unstable(feature = "unique", issue = "27730")]
+impl<T: fmt::Debug> fmt::Debug for Unique<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("Unique")
+            .field("pointer", &self.pointer)
+            .finish()
```

Comment:
maybe `Unique` and `Shared` should just print like a plain pointer? Somewhere in the guidelines we say that debug output doesn't need to contain type information.